src/firmware/main/shared/physics.h

```diff
@@ -114,6 +116,10 @@ void PolVel2Cart(float const loc[2], float vel[2]);
 void PolAcc2Cart(float const loc[2], float const vel[2], float const Pacc[2],
                  float Cacc[2]);
 
+// C++ does not allow variable length arguments, but we would still like to
+// enforce them when possible, so we compile this differently depending
+// on what language this header is being used in
+#ifndef __cplusplus
```

Comment:
Can you explain this?

Reply:
Ah, we can the one meant for c++ (the `else` case)`, but I think it was still a good option to make the array lengths clear like this when possible, so I left the C version in.

Reply:
seems like we can achieve the same clarity with some nice documentation

Reply:
I don't feel super strongly either way here, summoning @MathewMacDougall and @akhilveeraghanta  for another opinion

Reply:
I'm personally less concerned about the clarity and more concerned with whether or not the variable length arrays give us any other benefits.

The way we use the VLAs in our implementation doesn't seem to use any of the small additional benefits I'm aware of, such as automatically creating an array whose lifetime is managed in the scope of the function. (been looking at https://stackoverflow.com/questions/16672322/malloced-array-vs-variable-length-array). The only benefit we're getting is that the function signature is _slightly_ more explicit.

So for me it comes down to if I dislike `ifdefs` more than I dislike the lack of extra clarity. Assuming this file will be modified and updates a reasonable amount moving forwards, I dislike `ifdefs` more and therefore would prefer we switch everything to be compatible with both C and C++

Reply:
I agree as well, it is in the shared folder so it defeats the purpose having being compatible between C and C++ (similar to how constants.h is shared).

If this is specific to c++ then it doesn't belong in the shared folder anyway, so also in favor about removing `ifdefs`

Reply:
Great, seems like there is consensus here, removed the C-specific version!